### PR TITLE
Filters

### PR DESCRIPTION
**Pull request by santiago-rodrig**
# Changes

- **created** `filter.js` reducer and `changeFilter` action creator
- **created** `<CategoryFilter />` component to change store's filter value
- **added** filtering for `<BooksList />` component

If there is anything missing or wrong please let me know and I'll fix it as soon as possible.

Thanks for your time.

## P.D.

The application is lacking of style. I'll add styling at the end when I'm sure that elements, classes, and ids are not going to change anymore.

Student Success allowed me to work along for the rest of the React's section.

### CODE REVIEW

File: src/containers/CategoryFilter.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (shloch) [inline src/containers/CategoryFilter.js:2]: This PRESENTATIONAL component (like the other presentational components) should be place in the `src/components/` folder...please relocate it [see [requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/presentationalCategoryFilter.png)]

File: src/containers/BooksList.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+const Component = ({ books, filter, removeBook }) => {
+  const filteredBooks = () => {
+    const booksInArray = booksCollection(books);
+    if (filter === 'all') return booksInArray;
+    return booksInArray.filter(book => book.category === filter);
+  };
+  return (
+    <>
+      <CategoryFilter />
+      <table>
```
Comment (shloch) [inline src/containers/BooksList.js:36]: this component should implement `handleFilterChange` and passes it down to `<CategoryFilter />` component (as props) [see [requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/HandelChangeFilter.png)]

File: src/containers/CategoryFilter.js (hunk 1 above)
Comment (santiago-rodrig) [reply src/containers/CategoryFilter.js:2]: Maybe the instructions given in the path right were a bit ambiguous, I'll implement the changes immediately.

File: src/containers/BooksList.js (hunk 2 above)
Comment (santiago-rodrig) [reply src/containers/BooksList.js:36]: On it.

**Pull request merged**